File: 卡码/代码随想录/Java/src/com/kurumi/codeThoughts/贪心/_02_中等/LeetCode_0055.java
package com.kurumi.codeThoughts.贪心._02_中等;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 55. 跳跃游戏
 */
public class LeetCode_0055 {
    static class Solution {
        public boolean canJump(int[] nums) {
            int cover = 0;
            if (nums.length == 1)
                return true;
            for (int i = 0; i <= cover; i++) {
                cover = Integer.max(i + nums[i], cover);
                if (cover >= nums.length - 1)
                    return true;
            }
            return false;
        }
    }
}

File: 卡码/代码随想录/C++/贪心/02-中等/LeetCode_0055.cpp
//55. 跳跃游戏
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
	bool canJump(vector<int>& nums) {
		if(nums.size()==1) return true;
		int cover=0;
		for(int i=0;i<=cover;i++){
			cover=max(cover,i+nums[i]);
			if(cover>=nums.size()-1) return true;
		}
		return false;
	}
};

int main(){
  system("pause");
  return 0;
}
